system-addon/lib/NewTabInit.jsm

```diff
@@ -37,6 +37,11 @@ this.NewTabInit = class NewTabInit {
           this._queue.clear();
         }
         break;
+      case at.SEARCH_BOX_FOCUSED:
+        if (action._target.url === "about:home" && this.store.getState().Prefs.values["aboutHome.autoFocus"]) {
+          action._target.browser.focus();
```

Comment:
Oh duhh. I suppose we don't actually need to wait for the page to want focus we if know the page will want focus anyway. I.e., it's about:home. We didn't need this separate action, and instead we could just focus the browser immediately on NEW_TAB_INIT or some earlier action. Otherwise we have https://bugzilla.mozilla.org/show_bug.cgi?id=1400585

Reply:
oooh. right.

Reply:
I just did some initial testing, and NEW_TAB_INIT is still not early enough. Here are the frames that the location bar gets focus then loses focus (if we call `focus()` from NEW_TAB_INIT) then when the search box appears focused.
23 -> 31 -> 33 = 8 frames of wrong focus, 2 frames of no focus
28 -> 35 -> 39 = 7 frames of wrong focus, 4 frames of no focus
7 -> 14 -> 17 = 7 frames of wrong focus, 3 frames of no focus

Reply:
Although it is faster than waiting for SEARCH_BOX_FOCUSED:
2 -> 12 -> 21 = 10 frames of wrong focus, 9 frames of no focus
19 -> 30 -> 37 = 11 frames of wrong focus, 7 frames of no focus
4 -> 20 -> 26 = 16 frames of wrong focus, 6 frames of no focus

Seems like the fix needs to be on the Firefox side to have the focus not go to the location bar in the first place.